### PR TITLE
Add React Native interface

### PR DESCRIPTION
**Pull request by lencioni**
React Native does most of the heavy lifting for us, so this interface is
very simple.

### CODE REVIEW

File: src/interfaces/reactNativeInterface.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export default {
+  create(styleHash) {
```
Comment (spikebrehm) [inline src/interfaces/reactNativeInterface.js]: This could be a hash or an array, correct? Perhaps update the variable name to reflect this?

Comment (spikebrehm) [inline src/interfaces/reactNativeInterface.js]: Oh nevermind, I see how this is meant to be used.

Comment (lencioni) [inline src/interfaces/reactNativeInterface.js]: Right, this is what is called to prepare the styles so this is always an object.

File: src/interfaces/reactNativeInterface.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    return styleHash;
```
Comment (lelandrichardson) [inline src/interfaces/reactNativeInterface.js]: This should return `ReactNative.StyleSheet.create(styleHash)`, not `styleHash` directly.

Comment (lencioni) [inline src/interfaces/reactNativeInterface.js]: Oh sweet. I missed that. That makes sense.

Comment (lencioni) [inline src/interfaces/reactNativeInterface.js]: PR has been updated.

File: package.json [hunk 3]
```diff
@@ -12,7 +12,7 @@
     "check-only-changelog-changed": "(expr $(git status --porcelain 2>/dev/null| grep -v \"CHANGELOG.md\" | wc -l) >/dev/null && echo 'Only CHANGELOG.md may have uncommitted changes' && exit 1) || exit 0",
     "clean": "rimraf lib",
     "lint": "eslint .",
-    "mocha": "mocha --compilers js:babel-register,jsx:babel-register --require airbnb-js-shims --recursive test",
+    "mocha": "mocha --compilers js:babel-register,jsx:babel-register --require airbnb-js-shims --recursive test/_helpers --recursive test",
```
Comment (spikebrehm) [inline package.json]: Dang, `--recursive test` doesn't also include `test/_helpers/`?

File: test/_helpers/reactNativeMock.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    res[key] = index;
+    return res;
+  }, {});
+};
+
+// the cache key that real react native would get
+const key = require.resolve('react-native');
+
+// make sure the cache is filled with our lib
+require.cache[key] = {
```
Comment (spikebrehm) [inline test/_helpers/reactNativeMock.js]: Voodoo.

**Comment by spikebrehm** [comment]
Thanks, LGTM.

File: package.json (hunk 3 above)
Comment (ljharb) [inline package.json]: hm, you shouldn't need `--recursive` twice - and yes, it does include it, but for ordering purposes we need `_helpers` to be included first.

Comment (lencioni) [inline package.json]: Yeah, we need _helpers first. I'll remove the redundant flag.

**Pull request merged**